lib/lbt/resources/ResourceCollector.js

```diff
@@ -109,29 +109,39 @@ class ResourceCollector {
 	}
 
 	async enrichWithDependencyInfo(resourceInfo) {
-		return this._pool.getModuleInfo(resourceInfo.name).then((moduleInfo) => {
+		return this._pool.getModuleInfo(resourceInfo.name).then(async (moduleInfo) => {
 			if ( moduleInfo.name ) {
 				resourceInfo.module = moduleInfo.name;
 			}
 			if ( moduleInfo.dynamicDependencies ) {
 				resourceInfo.dynRequired = true;
 			}
-			if ( moduleInfo.dependencies.length > 0 ) {
-				resourceInfo.required = resourceInfo.required || new Set();
-				resourceInfo.condRequired = resourceInfo.condRequired || new Set();
-				moduleInfo.dependencies.forEach((dep) => {
-					if ( moduleInfo.isConditionalDependency(dep) ) {
-						resourceInfo.condRequired.add(dep);
-					} else if ( !moduleInfo.isImplicitDependency(dep) ) {
-						resourceInfo.required.add(dep);
-					}
-				});
-			}
+
+			this._applyModuleInfoDependenciesToResourceInfo(resourceInfo, moduleInfo);
+
 			if ( moduleInfo.subModules.length > 0 ) {
 				resourceInfo.included = resourceInfo.included || new Set();
-				moduleInfo.subModules.forEach((mod) => {
-					resourceInfo.included.add(mod);
-				});
+				await Promise.all(moduleInfo.subModules.map(async (subModule) => {
+					// Add subModule as included
+					resourceInfo.included.add(subModule);
+
+					// Try to inherit dependency info
+					let subModuleInfo;
+					try {
+						subModuleInfo = await this._pool.getModuleInfo(subModule);
+					} catch (err) {
+						log.verbose(`	missing submodule ${subModule} included by ${moduleInfo.name}`);
+					}
+					if (subModuleInfo) {
+						// Inherit dependencies
+						this._applyModuleInfoDependenciesToResourceInfo(resourceInfo, subModuleInfo);
```

Comment:
Dependencies should not be added if they are already listed as "included". I'm working on a fix..

Reply:
Also, a dependency should not be listed as required **and** condRequired